app/routes/circulars/circulars.server.ts

```diff
@@ -107,8 +107,68 @@ export async function search({
   totalItems: number
 }> {
   const client = await getSearch()
+  const isLuceneSyntax = query ? query.includes(':') : false
+  const sortOrder = query
+    ? { _score: 'desc' }
+    : { circularId: { order: 'desc' } }
 
   const [startTime, endTime] = getValidDates(startDate, endDate)
+  const esQuery = query
+    ? {
+        function_score: {
+          query: isLuceneSyntax
+            ? {
+                query_string: {
+                  query: `${query}~`,
+                  fuzziness: 'AUTO',
+                },
+              }
+            : {
+                bool: {
+                  must: [
+                    {
+                      query_string: {
+                        query: `*${query}*`,
+                        default_field: '*',
+                        fuzziness: 'AUTO',
+                      },
+                    },
+                    {
+                      range: {
+                        createdOn: {
+                          gte: startTime,
+                          lte: endTime,
+                        },
+                      },
+                    },
+                  ],
+                },
+              },
+          functions: [
+            {
+              filter: { wildcard: { eventId: `*${query}*` } },
+              weight: 4,
+            },
+            {
+              filter: { wildcard: { synonyms: `*${query}*` } },
+              weight: 3,
+            },
+            {
+              filter: { wildcard: { subject: `*${query}*` } },
+              weight: 2,
+            },
+            {
+              filter: { wildcard: { body: `*${query}*` } },
+              weight: 1,
+            },
+          ],
+          score_mode: 'sum',
+          boost_mode: 'sum',
+        },
```

Comment:
Are we sure that the Lucene query mode does not support boosted lists of fields, as in `"fields": ["title^4", "plot"]`? See https://opensearch.org/docs/latest/query-dsl/full-text/index/#multi-match.

Reply:
it's not because it's a lucene query, it's because it's a function_score. I could not use a function_score.

Reply:
What does `function_score` have to do with the `fields` property?

Reply:
nothing, the function_score has the functions property and is how it ranks results. that's why the ranking is done in functions here. I could remove the function score

Reply:
The `function_score` method seems more verbose than field boosting method. Can you use the field boosting method?